src/core/lib/security/credentials/tls/grpc_tls_crl_provider.h

```diff
@@ -0,0 +1,55 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPC_SRC_CORE_LIB_SECURITY_CREDENTIALS_TLS_GRPC_TLS_CRL_PROVIDER_H
+#define GRPC_SRC_CORE_LIB_SECURITY_CREDENTIALS_TLS_GRPC_TLS_CRL_PROVIDER_H
+
+#include <openssl/crypto.h>
+
+#include "absl/strings/string_view.h"
+
+#include <grpc/grpc_crl_provider.h>
+
+namespace grpc_core {
+namespace experimental {
+
+class CrlImpl : public Crl {
+ public:
+  explicit CrlImpl(X509_CRL* crl);
+  ~CrlImpl() override;
+  std::string Issuer() override;
+  X509_CRL& crl() const { return *crl_; }
+
+ private:
+  X509_CRL* crl_;
+  std::string issuer_;
+};
+
+class CertificateInfoImpl : public CertificateInfo {
+ public:
+  explicit CertificateInfoImpl(absl::string_view issuer) : issuer_(issuer) {}
+  absl::string_view GetIssuer() const override { return issuer_; }
+
+ private:
+  absl::string_view issuer_;
```

Comment:
This is meant to be a very short lived object that is created and immediately used to fetch a CRL, so we should control the lifetime of the underlying buffer in `ssl_transport_security.cc` and be certain that it's longer than the life of this object. It would be a single string copy to store it as a `std::string`.

WDYT?

Reply:
That's how we're using it right now, but I think that is dangerous because we may refactor the `ssl_transport_security.cc` code in the future or use this class for some other purpose in the future. I'd strongly suggest changing this. :)